### PR TITLE
CORCI-1074 build: Allow special CI targets

### PR DESCRIPTION
**Pull request by JohnMalmberg**
Separate out a 'ci_target' from a build 'target' to allow more easily
running tests on newer pre-built CI images.

dockerBuildArgs.groovy:
   Add CentOS 8 special app stream repo
   Fix missing variable declaration

parseStageInfo.groovy
   Set the 'ci_target' member for node provisioning

functionalTest.groovy:
scanRpms.groovy:
testRpm.groovy:
unitTest.groovy
   Now use 'ci_target' for node provisioning

provisionNodes.groovy:
   Fix some declarations to be more specific types
   Remove some dead code

Signed-off-by: John E Malmberg <john.e.malmberg@intel.com>

### CODE REVIEW

**Comment by JohnMalmberg** [comment]
DAOS test PR job of current master against these changes has less failures than the master build that it was based on.
https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-5622/92/

File: vars/functionalPackages.groovy [hunk 1]
```diff
@@ -30,8 +30,8 @@ String call(String distro, Integer client_ver, String next_version) {
         distro.startsWith('el7') || distro.startsWith('centos7') ||
         distro.startsWith('el8') || distro.startsWith('centos8') ||
         distro.startsWith('ubuntu20')) {
+        println("Functional Packages -${daos_pkgs}-")
```
Comment (brianjmurrell) [inline vars/functionalPackages.groovy:33]: Is this needed?  Can we drop if not?

File: vars/getDAOSPackages.groovy [hunk 2]
```diff
@@ -24,5 +24,7 @@ String call(String distro, String next_version) {
     if (distro.startsWith('ubuntu20')) {
         return pkgs + "=" + daosPackagesVersion(distro, next_version)
     }
+    String ret_str = pkgs + "-" + daosPackagesVersion(distro, next_version)
+    println("getDAOSPackages - pkgs=-${ret_str}")
```
Comment (brianjmurrell) [inline vars/getDAOSPackages.groovy:28]: Debug that was not removed?

File: vars/parseStageInfo.groovy [hunk 3]
```diff
@@ -200,7 +209,14 @@ def call(Map config = [:]) {
       result['test_tag'] += atag + ',' + cluster_size + ' '
     }
     result['test_tag'] = result['test_tag'].trim()
-  } // if (stage_name.contains('Functional'))
+    // if (stage_name.contains('Functional'))
+  } else if (stage_name.contains('Storage')) {
+    println("Nodelist = -${env.NODELIST}-")
```
Comment (brianjmurrell) [inline vars/parseStageInfo.groovy:214]: Debug to remove?

File: vars/provisionNodes.groovy [hunk 4]
```diff
@@ -90,13 +91,10 @@ def call(Map config = [:]) {
...
-  def inst_rpms = config.get('inst_rpms', '')
-  def inst_repos = config.get('inst_repos','')
+  String inst_rpms = config.get('inst_rpms', '')
+  String inst_repos = config.get('inst_repos','')
 
-  def repository_g = ''
-  def repository_l = ''
-
-  def gpg_key_urls = []
+  List gpg_key_urls = []
```
Comment (brianjmurrell) [inline vars/provisionNodes.groovy:97]: I don't think we actually use this GPG key URLs any more.  I could be wrong, but I seem to recall avoiding it's use in some refactoring recently.

File: vars/provisionNodes.groovy [hunk 5]
```diff
@@ -110,45 +108,10 @@ def call(Map config = [:]) {
...
-        if (env.DAOS_STACK_LEAP_15_LOCAL_REPO != null) {
-            repository_l = env.REPOSITORY_URL +
-                env.DAOS_STACK_LEAP_15_LOCAL_REPO
-        }
-    }
-  }
 
   if (!fileExists('ci/provisioning/log_cleanup.sh') ||
       !fileExists('ci/provisioning/post_provision_config.sh')) {
+      println('Falling back to old provisioning code')
```
Comment (brianjmurrell) [inline vars/provisionNodes.groovy:114]: Debug to be removed?

File: vars/storagePrepTest.groovy [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+   *                             Default determined by parseStageInfo().
+   *
+   */
+
+def call(Map config = [:]) {
+
+  println("Entering Storage Prep Test")
+  def nodelist = config.get('NODELIST', env.NODELIST)
+  def context = config.get('context', 'test/' + env.STAGE_NAME)
+  def description = config.get('description', env.STAGE_NAME)
```
Comment (brianjmurrell) [inline vars/storagePrepTest.groovy:62]: Lots of `def`s in this whole file.  Should they be more specific types?

File: vars/storagePrepTest.groovy [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+ 
+  println("Calling parseStage_info")
+  Map stage_info = parseStageInfo(config)
+
+  println("config = -${config}-")
```
Comment (brianjmurrell) [inline vars/storagePrepTest.groovy:67]: Lots of debug printing in here also.

File: vars/skipStage.groovy [hunk 8]
```diff
@@ -15,7 +15,8 @@ String skip_stage_pragma(String stage, String def_val='false') {
 }
 
 boolean skip_ftest(String distro) {
-    return distro == 'ubuntu20' ||
+    return env['CI_FUNCTIONAL_' + distro + '_TEST'] == 'false' ||
```
Comment (brianjmurrell) [inline vars/skipStage.groovy:18]: Parameters are also available as `params.<parameter-name>` such as is done here https://github.com/daos-stack/pipeline-lib/blob/faf1eb64ed7e8862bdaffe483ca0c1302bf401c6/vars/parseStageInfo.groovy#L166

I actually didn't know parameters were implicitly set into the environment.  I'm wondering about the pros/cons of using the environment hook for them vs the `params` object with regard to transparency about where they are actually coming from, etc.  I'm not sure I have a strong opinion, but `params` does seem more direct.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: vars/functionalPackages.groovy (hunk 1 above)
Comment (JohnMalmberg) [reply vars/functionalPackages.groovy:33]: Yeah, I missed some debug statements.  Will be gone in the next push.

File: vars/provisionNodes.groovy (hunk 4 above)
Comment (JohnMalmberg) [reply vars/provisionNodes.groovy:97]: It is being passed to the script that is being run.

File: vars/storagePrepTest.groovy (hunk 6 above)
Comment (JohnMalmberg) [reply vars/storagePrepTest.groovy:62]: Yes.

File: vars/skipStage.groovy (hunk 8 above)
Comment (JohnMalmberg) [reply vars/skipStage.groovy:18]: Changing to the params object for next push.

**Comment by JohnMalmberg** [comment]
https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-5622/111/ passed a regression test for this PR with DAOS

**Pull request merged**